src/whisper-flat.cpp

```diff
@@ -0,0 +1,60 @@
+#include "whisper.h"
+#include "whisper-arch.h"
+
+#include "ggml.h"
+#include "ggml-cpp.h"
+#include "ggml-alloc.h"
+#include "ggml-backend.h"
+
+#include <atomic>
+#include <algorithm>
+#include <cassert>
+#define _USE_MATH_DEFINES
+#include <cmath>
+#include <climits>
+#include <codecvt>
+#include <cstdarg>
+#include <cstdio>
+#include <cstring>
+#include <fstream>
+#include <functional>
+#include <map>
+#include <mutex>
+#include <random>
+#include <regex>
+#include <set>
+#include <string>
+#include <thread>
+#include <vector>
+
+#include "whisper-flat.h"
+
+#ifdef BINDINGS_FLAT
+void whisper_flat_backend_load_all(void) {
+    ggml_backend_load_all();
+}
+
+const char * whisper_flat_get_system_info_json(void) {
+    return whisper_get_system_info_json();
+}
+
+struct whisper_state * whisper_flat_get_state_from_context(struct whisper_context * ctx) {
```

Comment:
Should this really be exposed? My reason for asking is that `whisper_state` contains C++ constructs like `std::vector` and my understanding is that the purpose of this is avoid them as not all languages can deal with them in an easy way.

Reply:
Addressed in next reply - also whisper state is used in other functions - not just the ones I create (I'd have to look em up)